### ISSUE TITLE
Better Add entity dialog

### ISSUE THREAD

**Issue by jkiviluo** [spine-tools/Spine-Toolbox#2370]
Here's a mock-up for creating new entities. It was motivated by the need to support abstract classes (there is a need to show what is the subclass in order to distinguish entities with the same name but different subclass), but it should be an improvement for the current 'Add entities' dialog in other ways too. It limits the user choices based on the chosen entity in the Entity tree. It should also limit choices if specific entity class is chosen. If the whole tree is chosen, then it would be list of all possible entity_classes that can be filled in (but user should probably rather limit the choices).

![image](https://github.com/spine-tools/Spine-Toolbox/assets/40472544/0f7bd6e8-2927-48e7-ad16-27e67652eae4)

Naturally typing in would limit the choices in the drop-downs. 

Pressing enter should 
- 'stage' the current row (if fully filled)
- create a new row for the same entity_class
- move focus to the new row (and to the first column with choices to be made)

It would be nice if the dialog would update when the user chooses something else on the entity tree (keeping all new staged additions so far made in the view).

Links to https://github.com/spine-tools/Spine-Toolbox/issues/2364
and to https://github.com/spine-tools/SpineOpt.jl/issues/773

**Comment by DillonJ** [comment]
I was thinking that whatever we do here, you would need to be able to copy/paste. Also, currently, if I paste a relationship parameter value row into DB editor, it will create the relationship if it doesn't exist - this is very nice functionality that I use all the time

**Comment by jkiviluo** [comment]
Copy-pasting updated to the issue description.

> Also, currently, if I paste a relationship parameter value row into DB editor, it will create the relationship if it doesn't exist

I didn't know about that. I feel conflicted. It's of course very handy and I would probably have used it a lot had I known. However, it can be dangerous because you can paste in misspellings. Maybe the new entity name in the parameter value pane should be highlighted? Just to let user know that the entity was just created. Should be another issue...

**Comment by jkiviluo** [comment]
At least in v0.8 I can paste a new 0-D entity through the parameter value pane.

**Comment by manuelma** [comment]
Does it get added?

**Comment by manuelma** [comment]
What shouldn't work is paste a multi-D entities with non-existing elements. Only if the elements exist, the multi-D entity gets added on the fly.

**Comment by jkiviluo** [comment]
Yes, it got added. Does not work in 0.7 though. (Unknown entity 'foo')
